core/src/main/java/org/apache/iceberg/MergingSnapshotUpdate.java

```diff
@@ -217,22 +214,20 @@ protected void add(DataFile file) {
         // open all of the manifest files in parallel, use index to avoid reordering
         Tasks.range(filtered.length)
             .stopOnFailure().throwFailureWhenFinished()
-            .executeWith(getWorkerPool())
+            .executeWith(ThreadPools.getWorkerPool())
             .run(index -> {
-              ManifestFile manifest = filterManifest(
-                  deleteExpression, metricsEvaluator,
-                  manifests.get(index));
+              ManifestFile manifest = filterManifest(metricsEvaluator, manifests.get(index));
               filtered[index] = manifest;
             }, IOException.class);
 
-        for (ManifestFile manifest : filtered) {
-          PartitionSpec spec = ops.current().spec(manifest.partitionSpecId());
+        Arrays.stream(filtered).forEach(manifest -> {
```

Comment:
I'm not a fan of the streams API. It's okay, but I don't think it is as clear as using iterator in most cases. Is there a benefit to using it here? Is this required by Baseline?

Reply:
I'm using stream APIs in a bunch of places in order to reduce the [Cyclomatic Complexity.](http://checkstyle.sourceforge.net/config_metrics.html#CyclomaticComplexity) Cyclomatic Complexity is 12 here, thus placing a ceiling on the number of branch statements that are in the code. For loops, to my understanding, are considered as branching code that count towards the cyclomatic complexity. I'm open to exploring keeping for-each loops and looking at how we can reduce the cyclomatic complexity in other ways (e.g. more broken out methods).

Reply:
Never mind - breaking up the method into multiple smaller ones reduces the cyclomatic complexity enough.

Reply:
Thanks for the context. Let's break these into smaller methods to reduce the complexity.

I think what this change actually did was reduce the complexity in byte code (the block is now a lambda), while not actually solving the problem. The code is still looping here, where it was before. The code is just broken into a different object in byte code only.